### PR TITLE
Rework identity (again)

### PR DESCRIPTION
**Pull request by JAicewizard**
This completely re-does the identity example. This strips out a lot of the required knowledge like timers and just focuses on the `WidgetID` and how to use it.

This example just has 2 counters and 2 buttons. Each button increments the counter by one, but via commands and targets instead of mutating the data. This is once again very contrived but is much simpler then the original one.

### CODE REVIEW

File: druid/examples/identity.rs [hunk 1]
```diff
@@ -14,48 +14,34 @@
...
+//! This example is fairly contrived; the basic idea is that there are 2 counters,
+//! and 2 buttons. If you press button one, counter one goes up. If you press button
+//! two counter 2 goes up.
 //!
-//! The key insight is that each button is linked to a specific ColorWell, and
-//! can send messages that are only handled by that widget.
+//! The key insight is that each button is linked to a specific counter, and
+//! can send messages that are only handled by that widget/controler.
 //!
 //! This is a contrived example; if you were designing a real app you might
```
Comment (derekdreery) [inline druid/examples/identity.rs:24]:
```suggestion
//! If you were designing a real app you might
```

You already said this above.

File: druid/examples/identity.rs [hunk 2]
```diff
@@ -14,48 +14,34 @@
...
-//! choose a different mechanism (such as just representing all of this state
-//! in your `Data` type) but this is an example, and I couldn't think of anything
-//! better. ¯\_(ツ)_/¯
+//! choose a different mechanism (such as just just changin the `Data` in the on_click),
+//! but this is just an example.
```
Comment (derekdreery) [inline druid/examples/identity.rs:26]:
```suggestion
//! choose a different mechanism (such as just just changing the `Data` in the on_click).
//! Identities are useful when the source and target of the change are far apart in
//! the widget tree.
```

**Review by derekdreery — COMMENTED** [review]
Just a few suggestions regarding wording in the docs - I've not reviewed the code in this pass.

File: druid/examples/identity.rs [hunk 3]
```diff
@@ -14,48 +14,35 @@
...
 //!
-//! This example is fairly contrived; the basic idea is that there is one
-//! "top widget", displaying random colors, and are three rows of widgets,
-//! each containing a ColorWell and two buttons.
-//! One button 'pins' the ColorWell, sending it a color to display.
-//! The second button 'syncs' the ColorWell, which makes it start displaying
-//! the same random colors as the top widget.
+//! This example is fairly contrived; the basic idea is that there are 2 counters,
+//! and 2 buttons. If you press button one, counter one goes up. If you press button
+//! two counter 2 goes up.
```
Comment (cmyr) [inline druid/examples/identity.rs:19]:
```suggestion
//! This example is fairly contrived; the basic idea is that there are two counters,
//! and two buttons. If you press button one, counter one goes up. If you press button
//! two counter two goes up.
```

File: druid/examples/identity.rs [hunk 4]
```diff
@@ -14,48 +14,35 @@
...
 //!
-//! The key insight is that each button is linked to a specific ColorWell, and
-//! can send messages that are only handled by that widget.
+//! The key insight is that each button is linked to a specific counter, and
+//! can send messages that are only handled by that widget/controler.
 //!
```
Comment (cmyr) [inline druid/examples/identity.rs:23]:
```suggestion
//! The key insight is that the data is not mutated by the button directly;
//! instead the button sends a command to a separate controller widget
//! that performs the actual mutation.
//!
```

File: druid/examples/identity.rs [hunk 5]
```diff
@@ -14,48 +14,35 @@
...
-//! This is a contrived example; if you were designing a real app you might
-//! choose a different mechanism (such as just representing all of this state
-//! in your `Data` type) but this is an example, and I couldn't think of anything
-//! better. ¯\_(ツ)_/¯
+//! If you were designing a real app you might
+//! choose a different mechanism (such as just just changing the `Data` in the on_click).
+//! Identities are useful when the source and target of the change are far apart in
+//! the widget tree.
```
Comment (cmyr) [inline druid/examples/identity.rs:27]:
```suggestion
//! If you were designing a real app you might choose a different mechanism
//! (such as just just changing the `Data` in the on_click); however there are
//! other circumstances where widgets may need to communicate with specific
//! other widgets, and identity is a useful mechanism in those cases.
```

**Review by cmyr — APPROVED** [review]
Cool, I think this is a nice simplification. Will make some little tweaks and then merge. :)

**Pull request merged**